apis/v1alpha1/iam_request.go

```diff
@@ -0,0 +1,33 @@
+// Copyright 2023 The Authors (see AUTHORS file)
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package v1alpha1
+
+import resourcemanager "google.golang.org/api/cloudresourcemanager/v3"
+
+// IamRequest represents a request to update IAM policies.
+type IamRequest struct {
+	// List of ResourcePolicy, each specifies the IAM policy to be added for a
+	// GCP resource.
+	ResourcePolicies []*ResourcePolicy `json:"policies,omitempty"`
+}
+
+// ResourcePolicy specifies the IAM policy to be added for a GCP resource.
+type ResourcePolicy struct {
+	// Resource represents one of GCP organization, folder, and project.
+	Resource string `json:"resource,omitempty"`
+
+	// Policy contains a list of IAM members to role bindings.
+	Policy *resourcemanager.Policy `json:"policy,omitempty"`
```

Comment:
This will expose other fields in the policy, such as condition. I doubt we want to have that as user-provided input?

Reply:
Yes, if it is about confusing users we can add readme calling out that those fields will be ignored?

Reply:
I'm not sure why we need the whole `Policy` when we only care about role and members?

Reply:
Just why not use the existing type, but if you prefer the below alternative, I don't have strong opinion on this since it is simple enough.
```
type ResourcePolicy struct {
  Resource string 
  Bindings []*Binding 
}

type Binding struct {
  Members []string 
  Role string 
}
```

Reply:
> why not use the existing type

Because we only need a subset of it and we don't control the existing type - not even its document. The best document is the API (the contract) itself rather than a readme.

Since it's simple to replicate the binding, I'd prefer we do that.

Reply:
Updated.